dask/sizeof.py

```diff
@@ -215,3 +225,27 @@ def sizeof_pyarrow_table(table):
     @sizeof.register(pa.ChunkedArray)
     def sizeof_pyarrow_chunked_array(data):
         return int(_get_col_size(data)) + 1000
+
+
+def _register_entry_point_plugins(entry_points=None):
+    """Register sizeof implementations exposed by the entry_point mechanism."""
+    if entry_points is None:
+        entry_points = importlib_metadata.entry_points()
+
+    # Python 3.10+ / importlib_metadata >= 3.9.0
+    if hasattr(entry_points, "select"):
+        sizeof_entry_points = entry_points.select(group="dask.sizeof")
+    else:
+        sizeof_entry_points = entry_points.get("dask.sizeof", [])
```

Comment:
I think it's better to use `if sys.version_info` like [the pattern in dask/distributed]( https://github.com/dask/distributed/blob/608338516f7eb0e6f8205cdf70a579c3d4f92480/distributed/comm/registry.py#L10-L26) this way pyupgrade can remove the dead code when 3.9 is EOL